community/primitive-collections/src/main/java/org/neo4j/collection/primitive/hopscotch/PrimitiveLongObjectHashMap.java

```diff
@@ -165,4 +174,35 @@ public boolean equals( Object o )
             return hash == that.hash;
         }
     }
+
+    private class ValueIterable implements Iterable<VALUE>
+    {
+        private final PrimitiveLongObjectHashMap<VALUE> map;
+        private final PrimitiveLongIterator iterator;
+
+        ValueIterable( PrimitiveLongObjectHashMap<VALUE> map )
```

Comment:
It looks like this Iterable will only be able to produce one good iterator. Perhaps it's not super important, but the `private final PrimitiveLongIterator iterator` could might as well be created and kept inside the anonymous `Iterable` created in `iterator()`, couldn't it?